Developers/Writing_Bears.rst

```diff
@@ -276,3 +276,148 @@ parameter in ``HiddenResult`` should be the instance of the Bear that yields
 but it can be any object.
+
+More Configuration Options
+--------------------------
+coala provides settings to further configure your bear according to your needs.
+Here is the list of all the settings you can use:
+
+- `LANGUAGES`_
+- `REQUIREMENTS`_
+- `INCLUDE_LOCAL_FILES`_
+- `CAN_DETECT and CAN_FIX`_
+- `BEAR_DEPS`_
+- `Other Settings`_
+
+
+LANGUAGES
+~~~~~~~~~
+
+To indicate which languages your bear supports,
+you need to give it a `set` of strings as a value:
+
+.. code:: python
+
+    class SomeBear(Bear):
+        LANGUAGES = {'C', 'CPP','C#', 'D'}
+
+REQUIREMENTS
+~~~~~~~~~~~~
+
+To indicate the requirements of the bear,
+assign ``REQUIREMENTS`` a set with instances of 
+subclass of ``PackageRequirement`` such as:
+
+- PipRequirement
+- NpmRequirement
+- CondaRequirement
+- DistributionRequirement
+- GemRequirement
+- GoRequirement
+- JuliaRequirement
+- RscriptRequirement
+
+.. code:: python
+
+    class SomeBear(Bear):
+        REQUIREMENTS = {
+        PipRequirement('coala_decorators', '0.2.1')}
+
+To specify multiple requirements you can use
+the multiple method. This can receive both tuples of strings,
+in case you want a specific version, or a simple string,
+in case you want the latest version to be specified.
+
+.. code:: python
+
+    class SomeBear(Bear):
+        REQUIREMENTS = PipRequirement.multiple(
+            ('colorama', '0.1'),
+            'coala_decorators')
+
+INCLUDE_LOCAL_FILES
+~~~~~~~~~~~~~~~~~~~
+
+If your bear needs to include
+local files, then specify it by giving strings
+containing relative file paths to the
+``INCLUDE_LOCAL_FILES``.
+
+.. code:: python
+
+    class SomeBear(Bear):
+        INCLUDE_LOCAL_FILES = {'checkstyle.jar',
+            'google_checks.xml'}
+
+CAN_DETECT and CAN_FIX
+~~~~~~~~~~~~~~~~~~~~~
+
+To easily keep track of what a bear can do,
+you can set the value of `CAN_FIX` and
+`CAN_DETECT` sets.
+
+
+.. code:: python
+
+    class SomeBear(Bear):
+        CAN_DETECT = {'Unused Code', 'Spelling'}
+
+        CAN_FIX = {'Syntax', 'Formatting'}
+
+
+To view a full list of possible values, check this list:
+
+- `Syntax`
+- `Formatting`
+- `Security`
+- `Complexity`
+- `Smell`
+- `Unused Code`
+- `Redundancy`
+- `Variable Misuse`
+- `Spelling`
+- `Memory Leak`
+- `Documentation`
+- `Duplication`
+- `Commented Code`
+- `Grammar`
+- `Missing Import`
+- `Unreachable Code`
+- `Undefined Element`
+- `Code Simplification`
+
+Specifying something to `CAN_FIX` makes it obvious
+that it can be detected too, so it may be omitted
+from `CAN_DETECT`
+
+BEAR_DEPS
+~~~~~~~~~
+
+``BEAR_DEPS`` contains bear classes
+that are to be executed before this bear gets executed.
+The results of these bears will then be passed to the
+run method as a dict via the `dependency_results` argument.
+The dict will have the name of the Bear as key and the list
+of its results as value:
+
+.. code:: python
+
+    class SomeOtherBear(Bear):
+        BEAR_DEPS = {SomeBear}
```

Comment:
actually that passage is wrong isn't it? BEAR_DEPS is a constant not a function! :O can you add a commit to fix that? Or am I wrong... :thinking: at least this part should link there IMO